SDWebImage/UIImageView+WebCache.h

```diff
@@ -154,6 +155,30 @@
                   progress:(nullable SDWebImageDownloaderProgressBlock)progressBlock
                  completed:(nullable SDExternalCompletionBlock)completedBlock;
```

Comment:
I think this method should be annotated with `NS_REFINED_FOR_SWIFT` in case of naming collision. See #1797 and related issues.

Reply:
The generated interface are:

```swift
    open func sd_setImage(with url: URL?, placeholderImage placeholder: UIImage?, options: SDWebImageOptions = [], progress progressBlock: __ObjC.SDWebImageDownloaderProgressBlock?, completed completedBlock: __ObjC.SDExternalCompletionBlock? = nil)

    open func sd_setImage(with url: URL?, placeholderImage placeholder: UIImage?, options: SDWebImageOptions = [], transition: SDWebImageTransition?, progress progressBlock: __ObjC.SDWebImageDownloaderProgressBlock?, completed completedBlock: __ObjC.SDExternalCompletionBlock? = nil)
```

Can this cause issue ? I'll check one

Reply:
I tried and all of these two can be called. The previous name conflict issue is related to the `options` arguments, which contains a defualt value `[]`. But this new `transition` arg placed after the options, so there is no issue :)

Reply:
Isn't is kind weird that only `completed` was given default parameter `nil` but not `progressBlock`. Why is this happening?  Just out of curiosity

Reply:
@skyline75489 That is bridged by compiler. When your last arg is a `nullable block`, the generated Swift interface will be a optional block with the default value `nil`.(Which means user can omit this arg and this actually generate 2 interface)

And also, if you use Objective-C's `NS_OPTIONS`, the generated Swift interface is a `OptionSet` with the default value `[]`. So this arg can also be omit and actually generate 2 interface.

So, we may have 2 arg with default value, one is `options` and another is `completedBlock`. You can not mark anyother arg with a default value in Objective-C. All done by compiler :)
(You can write a simple demo to check, or contact me)